### PR TITLE
Making `dpp.rc` file

### PR DESCRIPTION
**Pull request by harshfeudal**
- When building, it will show more info for `dpp.dll` file (on Windows properties selection).
- Haven't tried to build yet because my pc crashed :( (testing help)

### CODE REVIEW

**Comment by harshfeudal** [comment]
- I want to make it like this btw:
image
- I decide to include `#include <winver.h>` but CMake thingy so :/

**Comment by SirObby** [comment]
Doesn't the rc file get generated by CMake?

**Comment by Axyte** [comment]
> @SirObby  : Doesn't the rc file get generated by CMake?

No.

**Comment by SirObby** [comment]
It does, I sent a screenshot in the discord.

**Comment by Axyte** [comment]
> It does, I sent a screenshot in the discord.

No, it does not. CMake generates a `.pc` file not a `.rc` file, read the extension carefully.

File: dpp.rc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+1 VERSIONINFO
+ FILEVERSION 10,0,16
```
Comment (braindigitalis) [inline dpp.rc:2]: this needs to be generated not hard coded in

File: dpp.rc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ FILESUBTYPE 0x0L
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040904b0"
+        BEGIN
+            VALUE "CompanyName", "Brainbox.cc"
+            VALUE "FileDescription", "Dpp - The C++ Discord API Library"
+            VALUE "FileVersion", "10.0.16"
+            VALUE "InternalName", "DppD"
```
Comment (braindigitalis) [inline dpp.rc:21]: what is DppD?

**Review by braindigitalis — CHANGES_REQUESTED** [review]
I can't see how this is even compiled to .res linked agsinst the executable

File: dpp.rc (hunk 2 above)
Comment (harshfeudal) [reply dpp.rc:21]: I mark `DppD` as Dpp DLL (yeah internal name only, it's just a name)

**Comment by braindigitalis** [comment]
how is this added to the lib? there's no changes to cmakelists?

**Comment by harshfeudal** [comment]
> how is this added to the lib? there's no changes to cmakelists?

About CMakeLists, I'm working on it and will soon pull on here when it works. I'm sorry for my latecy :(

**Comment by harshfeudal** [comment]
@braindigitalis I've seen you have provided this on your `CMakeLists.txt`:
image

**Comment by braindigitalis** [comment]
> @braindigitalis I've seen you have provided this on your `CMakeLists.txt`: image

I don't get it this is just a screenshot of something already in the cmakelists?

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**